marimo/_plugins/ui/_impl/table.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2023 Marimo. All rights reserved.
+from __future__ import annotations
+
+from typing import (
+    TYPE_CHECKING,
+    Callable,
+    Final,
+    List,
+    Literal,
+    Optional,
+    Sequence,
+    Union,
+    cast,
+)
+
+from marimo import _loggers
+from marimo._output.mime import MIME
+from marimo._output.rich_help import mddoc
+from marimo._plugins.ui._core.ui_element import UIElement
+
+LOGGER = _loggers.marimo_logger()
+
+Numeric = Union[int, float]
+
+if TYPE_CHECKING:
+    import pandas as pd
+
+TableData = Union[
+    list[dict[str, str]],
+    Sequence[str | int | float | bool | MIME | None],
+    Sequence[dict[str, str | int | float | bool | MIME | None]],
+    "pd.DataFrame",
+]
+
+
+@mddoc
+class table(UIElement[List[str], List[object]]):
+    """
+    A table component.
+
+    **Example.**
+
+    ```python
+    table = mo.ui.table(
+      data=[
+        {'first_name': 'Michael', 'last_name': 'Scott'},
+        {'first_name': 'Dwight', 'last_name': 'Schrute'}
+      ],
+      label='Users'
+    )
+    ```
+
+    ```python
+    # df is a Pandas dataframe
+    table = mo.ui.table(
+        data=df.to_dict('records'),
+        # use pagination when your table has many rows
+        pagination=True,
+        label='Dataset'
+    )
+    ```
+
+    **Attributes.**
+
+    - `value`: the selected values, or `None` if no selection.
+    - `data`: the table data
+
+    **Initialization Args.**
+
+    - `data`:  a list of values representing a column, or a list of dicts
```

Comment:
update doc to mention that `data` can be a dataframe? I'd start with that.

"A pandas dataframe, or a list of values representing a column, or a list of dicts ...